cpp/src/arrow/util/future_test.cc

```diff
@@ -122,15 +125,19 @@ void AssertFinished(const Future<T>& fut) {
 // Assert the future is successful *now*
 template <typename T>
 void AssertSuccessful(const Future<T>& fut) {
-  ASSERT_EQ(fut.state(), FutureState::SUCCESS);
-  ASSERT_OK(fut.status());
+  if (IsFutureFinished(fut.state())) {
```

Comment:
This is changing the meaning of the tests, as now a pending future will pass this silently.

Reply:
Good catch.  There should be an else block calling FAIL.  I'll add that.  The intent of this "if" wrapper is to prevent a pending future from deadlocking the test (which makes it difficult to debug).  It absolutely should fail if the future is pending.